### PR TITLE
Fix noise and rerun when removing overlap

### LINKED ISSUE #265: fix coupled_noise when removed overlap
When removing some overlap, we have to make sure that `w2s2` is 0 in the removed area.

### PR DESCRIPTION
**Pull request by carlosggarcia**
Closes #265, #266 and similar bugs. Fixes the noise and the catalog rerun paths when removing the overlap.

### CODE REVIEW

File: xcell/mappers/mapper_DELS.py [hunk 1]
```diff
@@ -85,7 +85,9 @@ def get_catalog(self):
             catalog (Array)
         """
         if self.cat_data is None:
-            fn = f'{self.map_name}_cat.fits'
+            # This will handle the situation when removing the overlap
+            map_name = self.map_name.split("_")[0] + f"_bin{self.zbin}"
+            fn = f'{map_name}_cat.fits'
```
Comment (damonge) [inline xcell/mappers/mapper_DELS.py:90]: Not sure why you need to do this for DELS. This seems to be cutting it in redshift?

File: xcell/mappers/mapper_HSC_DR1wl.py [hunk 2]
```diff
@@ -118,7 +118,9 @@ def get_catalog(self):
             cat (Array)
         """
         if self.cat is None:
-            fn = f'{self.map_name}.fits'
+            # This will handle the situation when removing the overlap
+            map_name = self.map_name.split("_")[0] + f"_{self.bn}"
+            fn = f'{map_name}.fits'
```
Comment (damonge) [inline xcell/mappers/mapper_HSC_DR1wl.py:123]: same question here as in DELS

File: xcell/mappers/mapper_KV450.py [hunk 3]
```diff
@@ -70,7 +70,9 @@ def get_catalog(self):
             array: catalog data
         """
         if self.cat_data is None:
-            fn = f'{self.map_name}_cat.fits'
+            # This will handle the situation when removing the overlap
+            map_name = self.map_name.split("_")[0] + f"_bin{self.zbin}"
+            fn = f'{map_name}_cat.fits'
```
Comment (damonge) [inline xcell/mappers/mapper_KV450.py:75]: As above

File: xcell/mappers/mapper_KiDS1000.py [hunk 4]
```diff
@@ -72,7 +72,9 @@ def get_catalog(self):
             cat_data (Array)
         """
         if self.cat_data is None:
-            fn = f'{self.map_name}_cat.fits'
+            # This will handle the situation when removing the overlap
+            map_name = self.map_name.split("_")[0] + f"_bin{self.zbin}"
+            fn = f'{map_name}_cat.fits'
```
Comment (damonge) [inline xcell/mappers/mapper_KiDS1000.py:77]: same

File: xcell/mappers/mapper_WIxSC.py [hunk 5]
```diff
@@ -105,7 +105,9 @@ def _get_catalog(self):
 
     def get_catalog(self):
         if self.cat_data is None:
-            fn = f'{self.map_name}_rerun_coord{self.coords}.fits'
+            # This will handle the situation when removing the overlap
+            map_name = self.map_name.split("_")[0] + f"_bin{self.zbin}"
+            fn = f'{map_name}_rerun_coord{self.coords}.fits'
```
Comment (damonge) [inline xcell/mappers/mapper_WIxSC.py:110]: and again

File: xcell/mappers/mapper_DELS.py (hunk 1 above)
Comment (carlosggarcia) [reply xcell/mappers/mapper_DELS.py:90]: Yep, that's how it was before too. `self.map_name` was `DELS_binX`. So here I'm just keeping this structure by removing the part from the removed overlap. 

Same in the others

Comment (damonge) [reply xcell/mappers/mapper_DELS.py:90]: I don't understand. So `split("_")[0]` is meant to remove some part of the map name that denotes that you've cut some area?

Comment (carlosggarcia) [reply xcell/mappers/mapper_DELS.py:90]: That is correct. This is appended in `MapperBase` https://github.com/xC-ell/xCell/blob/9e6e4ff1e8c6a0fe2641dff5ae1a12400fa76406/xcell/mappers/mapper_base.py#L39-L43

Comment (damonge) [reply xcell/mappers/mapper_DELS.py:90]: oh, wow, that's very niche! OK then.

**Review by damonge — APPROVED** [review]

**Pull request merged**